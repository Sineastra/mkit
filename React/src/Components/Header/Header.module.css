.header {
  z-index: 1;
  max-width: 100vw;
  position: relative;
  padding-left: 20px;
  padding-right: 20px;
  height: 150px;
  background-color: rgba(255, 255, 255, 0.2);
  border-bottom: 1px solid black;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  align-content: initial;
}
.header > a {
  color: var(--theme-text-color);
  text-decoration: none;
}
.header > form {
  width: 500px;
  height: 60px;
}

/*# sourceMappingURL=Header.module.css.map */
