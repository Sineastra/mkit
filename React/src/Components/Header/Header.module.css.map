{"version":3,"sourceRoot":"","sources":["Header.module.scss","../../_scssPartials/_variables.scss","../../_scssPartials/_mixins.scss"],"names":[],"mappings":"AAGA;EACI;EACA;EACA;EACA;EACA;EACA,QCTY;EDUZ;EACA;EEVA;EACA,gBFUqB;EETrB,iBFS0B;EER1B,aFQyC;EEPzC,eALwF;;AFcxF;EACI;EACA;;AAGJ;EACI;EACA","file":"Header.module.css"}