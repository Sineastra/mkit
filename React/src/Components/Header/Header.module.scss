@import "../../_scssPartials/mixins";
@import "../../_scssPartials/variables";

.header {
    z-index: 1;
    max-width: 100vw;
    position: relative;
    padding-left: 20px;
    padding-right: 20px;
    height: $header-height;
    background-color: rgba(255, 255, 255, 0.2);
    border-bottom: 1px solid black;
    @include displayFlex(row, space-between, center);

    & > .logoWrapper {
        @include displayFlex(column, center);

        & > a {
            color: var(--theme-text-color);
            text-decoration: none;
        }

        & > .logout {
            margin: 0;
            cursor: pointer;
        }
    }

    & > form {
        width: 500px;
        height: 60px;
    }
}