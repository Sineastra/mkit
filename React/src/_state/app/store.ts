import { configureStore } from "@reduxjs/toolkit"
import userDataReducer from "../features/userSlice"
import notifDataReducer from "../features/notifSlice"


const store = configureStore({
	reducer: {
		userData: userDataReducer,
		notifData: notifDataReducer,
	},
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export default store