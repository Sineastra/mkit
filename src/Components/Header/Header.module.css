.header {
  z-index: 1;
  width: 100vw;
  position: relative;
  padding-left: 20px;
  padding-right: 20px;
  height: 150px;
  background-color: rgba(255, 255, 255, 0.2);
  font-family: "Comfortaa", cursive;
  border-bottom: 1px solid black;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  align-content: initial;
}
.header > h1 {
  color: var(--theme-text-color);
}
.header > form {
  width: 500px;
  height: 60px;
}

/*# sourceMappingURL=Header.module.css.map */
