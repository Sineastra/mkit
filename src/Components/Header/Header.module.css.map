{"version":3,"sourceRoot":"","sources":["Header.module.scss","../../_scssPartials/mixins/_mixins.scss"],"names":[],"mappings":"AAEA;EACI;EACA;EACA;EACA;EACA;EACA;ECPA;EACA,gBDOqB;ECNrB,iBDM0B;ECL1B,aDKyC;ECJzC,eALwF;;ADWxF;EACI;;AAGJ;EACI;EACA","file":"Header.module.css"}