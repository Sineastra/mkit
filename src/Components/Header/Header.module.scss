@import "../../_scssPartials/mixins/mixins";
@import "../../_scssPartials/variables";

.header {
    z-index: 1;
    width: 100vw;
    position: relative;
    padding-left: 20px;
    padding-right: 20px;
    height: $header-height;
    background-color: rgba(255, 255, 255, 0.2);
    font-family: 'Comfortaa', cursive;
    border-bottom: 1px solid black;
    @include displayFlex(row, space-between, center);

    & > h1 {
        color: var(--theme-text-color);
    }

    & > form {
        width: 500px;
        height: 60px;
    }
}