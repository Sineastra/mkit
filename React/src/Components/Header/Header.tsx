import SearchBar from "../SearchBar/SearchBar"
import styles from "./Header.module.scss"
import { Link } from "react-router-dom"
import { useSelector } from "react-redux"
import { RootState } from "../../_state/app/store"


const Header = () => {
	const user = useSelector((state: RootState) => state.userData)

	return (
		user.loggedUser &&
		<header className={ styles.header }>
			<Link to="/">
				<h1>{ user.loggedUser.name }'s Collection</h1>
			</Link>
			<SearchBar/>
		</header>
	)
}

export default Header