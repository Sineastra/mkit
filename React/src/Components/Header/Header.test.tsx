import renderer from "react-test-renderer"
import Header from "./Header"
import { BrowserRouter } from "react-router-dom"
import { Provider } from "react-redux"
import store from "../../_state/app/store"


describe("---> Testing /Components/Header", () => {
	it("snapshot test", () => {
		const tree = renderer.create(
			<Provider store={ store }>
				<BrowserRouter>
					<Header/>,
				</BrowserRouter>,
			</Provider>,
		).toJSON()

		expect(tree).toMatchSnapshot()
	})
})

export {}