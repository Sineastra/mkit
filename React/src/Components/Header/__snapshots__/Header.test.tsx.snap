// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`---> Testing /Components/Header snapshot test 1`] = `
Array [
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        My Movie Collection
      </h1>
    </a>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search by movie title..."
        type="search"
        value=""
      />
      <button>
        Search
      </button>
    </form>
  </header>,
  ",",
]
`;
